assemblyline/filestore/transport/ftp.py

```diff
@@ -215,3 +214,31 @@ def put(self, dst_path, content):
             self.log.debug("Rename: %s -> %s", temppath, finalpath)
             self.ftp.rename(temppath, finalpath)
             assert (self.exists(dst_path))
+
+    def read(self, path):
+        path = self.normalize(path)
+        callbackqueue = queue.Queue()
+        def retrbinary(chunk_size = 8192):
+            self.ftp.retrbinary('RETR ' + path, callback = callbackqueue.put, blocksize = chunk_size)
+            callbackqueue.put(None)
+        return TransportReadStreamFTP(path, callbackqueue, retrbinary)
+
+
+class TransportReadStreamFTP(TransportReadStream):
+    def __init__(self, filepath, filequeue, retrMethod):
+        self.filepath = filepath
+        self.filequeue = filequeue
+        self.retrMethod = retrMethod
+
+    def close(self):
+        pass
+
+    def read(self, chunk_size = 8192):
+        if self.readthread is None:
+            self.readthread = threading.Thread(target=self.retrMethod(chunk_size), daemon=True)
+            self.readthread.start()
+        chunk = self.filequeue.get()
```

Comment:
Make sure read method doesn't get blocked forever if there is an extra call to read after the last one that returns none